edweiss-app/app/(app)/deck/[id]/index.tsx

```diff
@@ -1,36 +1,249 @@
-			</TView>
+					<FancyButton
+						backgroundColor='blue'
+						onPress={cancelCardSelection}
+						style={{ flex: 1 }}
+					>
+						Cancel
+					</FancyButton>
+				</TView>
+			)}
+
+			< TScrollView >
+
+				{/*goToPath={() => router.push({ pathname: `deck/${id}/card` as any, params: { deckId: id, cardIndex: cards.indexOf(card) } })}*/}
+				<For each={sortedCards}>
+					{(card, index) => (
+						<DisplayCard
+							key={sortedCards.indexOf(card)}
+							card={card}
+							isSelected={selectedCards.some(selected => selected.question === card.question)}
+							toggleSelection={toggleCardSelection}
+							onLongPress={enterSelectionMode}
+							selectionMode={selectionMode}
+							goToPath={() => handleCardPress(index)}
+						/>
+					)}
+				</For>
+
+			</TScrollView >
+
+			{/* Modal to display the CardScreenComponent */}
+			<ModalContainer modalRef={modalRef} snapPoints={['50%', '90%']}>
+				{selectedCardIndex !== null &&
+					<CardScreenComponent deckId={id} cardIndex={cards.indexOf(sortedCards[selectedCardIndex])} isModal={true} />
+				}
+
+			</ModalContainer >
+
+			{/* Buttons for navigation */}
+			< TView >
+				<FancyButton icon='play' onPress={() => router.push({ pathname: `deck/${id}/playingCards` as any })}>
+					Go to Playing screen
+				</FancyButton>
+
+				<FancyButton mt={'md'} mb={'sm'} ml={'md'} mr={'md'} textColor='crust' onPress={() => router.push({ pathname: `/deck/${id}/card/creation` as any, params: { deckId: id } })} icon='create-outline' >
+					Create Card
+				</FancyButton>
+			</TView >
 		</>
 	);
 };
 
-export default Deck;
+export default CardListScreen;
+
+const DisplayCard: ReactComponent<{ card: Memento.Card, isSelected: boolean, toggleSelection: (card: Memento.Card) => void; onLongPress: () => void; selectionMode: boolean; goToPath: () => void; }> = ({ card, isSelected, toggleSelection, onLongPress, selectionMode, goToPath }) => {
+	// Determine the text color based on the learning status
+	const statusColor = getStatusColor(card.learning_status ?? "");
+
+	const handlePress = () => {
+		if (!selectionMode) {
+			goToPath();
+		} else {
+			toggleSelection(card); // Select or deselect
+		}
+	};
+
+	return (
+		<TTouchableOpacity bb={5}
+			onLongPress={() => {
+				onLongPress();
+				toggleSelection(card);
+			}}
+			onPress={handlePress}
+			m='md' mt={'sm'} mb={'sm'} p='lg'
+			backgroundColor={isSelected ? 'rosewater' : 'base'}
+			borderColor='crust' radius='lg'
+		>
+			<TText bold>
+				{card.question}
+			</TText>
+			<TText mb='md' color='subtext0' size={'sm'}>
+				2h
+			</TText>
+			{/* Learning Status Display */}
+			<TText style={{
+				position: 'absolute',
+				top: 25,
+				right: 10,
+				backgroundColor: 'lightgray',
+				padding: 8,
+				borderRadius: 5,
+				color: statusColor
+			}} size={'sm'}>
+				{card.learning_status}
+			</TText>
+			{isSelected && <TText color='green'>✓</TText>}
+		</TTouchableOpacity>
+	);
+};
+
+function getStatusColor(status: string) {
```

Comment:
Maybe consider doing a Record, (you can check it out in the app on utils/todo/utilsFunctions.tsx). However, it a matter of tastes, this does the job just fine.

Reply:
Got it :D